core/wiki/macros/toc.tid

```diff
@@ -1,218 +1,422 @@
-            <$transclude tiddler=<<toc-closed-icon>> />
-            </$button>
-          </$reveal>
-          <$reveal type="match" stateTitle=<<toc-state>> text="open">
-            <$button setTitle=<<toc-state>> setTo="close" class="tc-btn-invisible tc-popup-keep">
-              <$transclude tiddler=<<toc-open-icon>> />
-            </$button>
-          </$reveal>
-        </$list>
-        <<toc-caption>>
-      </$link>
-      <$reveal type="match" stateTitle=<<toc-state>> text="open">
-        <$macrocall $name="toc-selective-expandable" tag=<<currentTiddler>> sort=<<__sort__>> itemClassFilter=<<__itemClassFilter__>> exclude=<<__exclude__>> path=<<__path__>>/>
-      </$reveal>
-    </li>
-  </$set>
+\procedure toc-unlinked-expandable-body(tag,sort:"",itemClassFilter,exclude,path,captionField,tagField,parentField,itemTemplate)
+<$qualify name="toc-state" title=<<tf.toc-stateTitle>>>
+	<li class=<<tf.toc-itemClass>>>
+		<<toc-item isUnlinked:"yes">>
+		<!-- TODO auto expand -->
+		<% if [<toc-state>get[text]match[open]] %>
+			<$macrocall $name="toc-expandable"
+				tag=<<currentTiddler>> sort=<<sort>> exclude=<<exclude>> path=<<path>>
+				captionField=<<captionField>> tagField=<<tagField>> parentField=<<parentField>>
+				itemTemplate=<<itemTemplate>>
+			/>
+		<% endif %>
+	</li>
 </$qualify>
 \end
 
-\define toc-unlinked-selective-expandable-body(tag,sort:"",itemClassFilter,exclude,path)
-\whitespace trim
-<$qualify name="toc-state" title={{{ [[$:/state/toc]addsuffix<__path__>addsuffix[-]addsuffix<currentTiddler>] }}}>
-  <$set name="toc-item-class" filter=<<__itemClassFilter__>> emptyValue="toc-item-selected" value="toc-item">
-    <li class=<<toc-item-class>>>
-      <$list filter="[all[current]tagging[]$sort$limit[1]] -[subfilter<__exclude__>]" variable="ignore" emptyMessage="""<$button class="tc-btn-invisible">{{$:/core/images/blank}}</$button><span class="toc-item-muted"><<toc-caption>></span>""">
-        <$reveal type="nomatch" stateTitle=<<toc-state>> text="open">
-          <$button setTitle=<<toc-state>> setTo="open" class="tc-btn-invisible tc-popup-keep">
-            <$transclude tiddler=<<toc-closed-icon>> />
-            <<toc-caption>>
-          </$button>
-        </$reveal>
-        <$reveal type="match" stateTitle=<<toc-state>> text="open">
-          <$button setTitle=<<toc-state>> setTo="close" class="tc-btn-invisible tc-popup-keep">
-            <$transclude tiddler=<<toc-open-icon>> />
-            <<toc-caption>>
-          </$button>
-        </$reveal>
-      </$list>
-      <$reveal type="match" stateTitle=<<toc-state>> text="open">
-        <$macrocall $name="toc-selective-expandable" tag=<<currentTiddler>> sort=<<__sort__>> itemClassFilter=<<__itemClassFilter__>> exclude=<<__exclude__>> path=<<__path__>>/>
-      </$reveal>
-    </li>
-  </$set>
+<!-- ==========================================================
```

Comment:
This is another non-standard new type of marker. We should be consistent.

Reply:
Yes. I think we should introduce something like this when we update core macros. It makes it much easier to visually separate the main macros from helper functions.